### PR TITLE
Add workload status to unit details

### PR DESCRIPTION
**Pull request by anthonydillon**

### CODE REVIEW

**Comment by anthonydillon** [comment]
## QA
- Set up GIJoe
- Deploy some services
- When there pending go into the unit list for that service and see that workload states 
- Click on the unit to go to the unit details and see the status is also displayed there
- Check the status is updated as services changes state

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5409/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5410/
Test PASSed.

File: jujugui/static/gui/src/app/components/check-list-item/_check-list-item.scss [hunk 1]
```diff
@@ -35,6 +36,18 @@
         }
     }
 
+    &--extra-info {
+      .check-list-item__label {
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/check-list-item/_check-list-item.scss]: I think we have a 4 space indent for the css.

File: jujugui/static/gui/src/app/components/check-list-item/check-list-item.js [hunk 2]
```diff
@@ -43,11 +44,26 @@ YUI.add('check-list-item', function() {
...
     },
 
     /**
+      Display the extraInfo if there is a status message
+
+      @method _getExtraInfo
+      @param {String} status The unit message.
+      @returns {String} The formatted status.
+    */
+    _getExtraInfo: function(info) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/check-list-item/check-list-item.js]: This function isn't used.

File: jujugui/static/gui/src/app/components/unit-details/test-unit-details.js [hunk 3]
```diff
@@ -59,19 +59,48 @@ describe('UnitDetails', function() {
         unit={fakeUnit} />);
 
     assert.deepEqual(output.props.children[0],
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/unit-details/test-unit-details.js]: You're deep checking that `output.props.children[0]` deep equals itself here :)

File: jujugui/static/gui/src/app/components/unit-details/test-unit-details.js [hunk 4]
```diff
@@ -59,19 +59,48 @@ describe('UnitDetails', function() {
...
+      <juju.components.UnitDetails
+        acl={acl}
+        changeState={sinon.stub()}
+        destroyUnits={sinon.stub()}
+        service={service}
+        previousComponent='units'
+        unitStatus='Installing software'
+        unit={fakeUnit} />);
+
+    assert.deepEqual(output.props.children[0],
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/unit-details/test-unit-details.js]: And here

File: jujugui/static/gui/src/app/components/check-list-item/check-list-item.js [hunk 5]
```diff
@@ -116,6 +132,17 @@ YUI.add('check-list-item', function() {
       }
     },
 
+    _generateExtraInfo: function(extraInfo) {
+      if (!extraInfo || extraInfo === '') {
+        return;
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/check-list-item/check-list-item.js]: If you return null here, does it stop requiring the `{undefined}` in the tests?

File: jujugui/static/gui/src/app/components/unit-details/test-unit-details.js (hunk 3 above)
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/unit-details/test-unit-details.js]: I copied this from other tests in this file. The code base is littered with these? Are they not right?

File: jujugui/static/gui/src/app/components/check-list-item/check-list-item.js (hunk 2 above)
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/check-list-item/check-list-item.js]: It's gone.

File: jujugui/static/gui/src/app/components/check-list-item/_check-list-item.scss (hunk 1 above)
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/check-list-item/_check-list-item.scss]: Done, thanks.

File: jujugui/static/gui/src/app/components/check-list-item/check-list-item.js (hunk 5 above)
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/check-list-item/check-list-item.js]: That puts a null into the element so think its best to leave it consistent with other components

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5412/
Test PASSed.

File: jujugui/static/gui/src/app/components/unit-details/test-unit-details.js (hunk 3 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/unit-details/test-unit-details.js]: It's unlikely, the call signature for deepEquals is 
`assert.deepEqual(actual, expected, message)`
You're comparing `output.props.children[0]` to itself and then have some jsx as the message if it's not

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5413/
Test PASSed.

**Comment by hatched** [comment]
Thanks @anthonydillon Looks good! :+1:

**Comment by hatched** [comment]
QA :ok: 

It's nice having those statuses there!

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**